src/smt/theory_str_noodler/theory_str_noodler.cpp

```diff
@@ -1038,111 +1035,156 @@ namespace smt::noodler {
         }
     }
 
-    zstring theory_str_noodler::model_of_string_expr(app* str_expr) {
-        // function that returns either the length of str var or model of int var from arith_model
-        std::function<rational(BasicTerm)> get_arith_model_of_var = [this](BasicTerm var) -> rational {
-            SASSERT(arith_model != nullptr);
-            expr_ref arg(m);
-            // the following is similar to code in util::len_to_expr()
-            if(!var_name.contains(var)) {
-                // if the variable is not found, it was introduced in the preprocessing/decision procedure
-                // (either as a string or int var), i.e. we can just create a new z3 variable with the same name 
-                arg = util::mk_int_var(var.get_name().encode(), m, m_util_a);
-            } else {
-                arg = var_name.at(var); // for int vars, we just take the var
-                if (m_util_s.is_string(arg->get_sort())) {
-                    // for string vars, we want its length
-                    arg = expr_ref(m_util_s.str.mk_length(arg), m);
+    class theory_str_noodler::noodler_var_value_proc : public model_value_proc {
```

Comment:
You mean only the class? Or the whole thing with model generation (i.e. also functions `mk_value` etc.)? Because these classes should be close to these functions I think in the code and they are declared as part of the class `theory_str_noodler`.

Reply:
> Also some comment of the class would be helpful.

I added some comments

Reply:
I suggest to move all 3 classes and the model generation functions to a separate file. In a similar way as we have done it with final check stuff.

Reply:
I moved them to a new file + I also finally moved `final_check_eh()` to the proper file.